lib/ansible/inventory/manager.py

```diff
@@ -161,13 +165,59 @@ def __init__(self, loader, sources=None, parse=True, cache=True):
         else:
             self._sources = sources
 
+        self._launch_ssh_agent()
+
         # get to work!
         if parse:
             self.parse_sources(cache=cache)
 
         self._cached_dynamic_hosts = []
         self._cached_dynamic_grouping = []
 
+    def _launch_ssh_agent(self):
+        ssh_agent_cfg = C.config.get_config_value('SSH_AGENT')
+        match ssh_agent_cfg:
+            case 'none':
+                return
+            case 'auto':
+                ssh_agent_dir = os.path.join(C.DEFAULT_LOCAL_TMP, 'ssh_agent')
+                os.mkdir(ssh_agent_dir, 0o700)
+                try:
+                    ssh_agent_bin = get_bin_path('ssh-agent', required=True)
+                except ValueError as e:
+                    raise AnsibleError('SSH_AGENT set to auto, but cannot find ssh-agent binary') from e
+                sock = os.path.join(ssh_agent_dir, 'agent.sock')
+                p = subprocess.Popen(
```

Comment:
Can we wrap this with exception handler?

Reply:
Wouldn't that need a `check=True`?

Reply:
ValueError and OSError would sufficient, but I would refer for more info - https://docs.python.org/3/library/subprocess.html#exceptions

Reply:
I don't think we should handle a `ValueError` here. It says it's raised on invalid arguments. So the arguments just should be of correct types. But `OSError` is a good idea.